webhook_to_fedora_messaging/auth.py

```diff
@@ -0,0 +1,52 @@
+from fastapi import Depends, HTTPException, Security
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_403_FORBIDDEN
+
+from webhook_to_fedora_messaging.database import get_session
+from webhook_to_fedora_messaging.models import User
+
+
+def user_factory(optional: bool = False, **kwargs):
+    """
+    Factory creating FastAPI dependencies for authenticating users
+    """
+    if optional:
+        kwargs["auto_error"] = False
+    security = HTTPBasic(realm="W2FM", **kwargs)
+
+    async def user_actual(
+        session: AsyncSession = Depends(get_session),  # noqa : B008
+        cred: HTTPBasicCredentials = Security(security)  # noqa : B008
+    ):
+        if not cred:
```

Comment:
My IDE is telling me that this can never happen. Can `cred` be `None`, or is it covered by the `HTTPBasic` dependency?

Reply:
It is. 

![image](https://github.com/user-attachments/assets/ef3e6d74-dcfc-4374-99e6-b6b0a14a8e0b)

Look at the `Authorize` button popping up there.

It understands it.